### PR TITLE
Core: Warn & fill jQuery.trim

### PR DESCRIPTION
**Pull request by mgol**
Fixes gh-325

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -60,15 +61,15 @@ for ( findProp in oldFind ) {
 }
 
 // The number of elements contained in the matched element set
-jQuery.fn.size = function() {
-	migrateWarn( "jQuery.fn.size() is deprecated and removed; use the .length property" );
+migrateWarnFunc( jQuery.fn, "size", function() {
```
Comment (mgol) [inline src/core.js]: This is not related but I thought I might as well use the helper we have

File: test/core.js [hunk 2]
```diff
@@ -370,15 +370,63 @@ QUnit.test( "jQuery.holdReady (warn only)", function( assert ) {
 	} );
 } );
 
+QUnit[ jQueryVersionSince( "3.1.1" ) ? "test" : "skip" ]( "jQuery.trim", function( assert ) {
+	assert.expect( 26 );
+
+	var nbsp = String.fromCharCode( 160 );
+
+	expectWarning( assert, "jQuery.trim", 1, function() {
```
Comment (mgol) [inline test/core.js]: I split it like that as Migrate now only warns once against any API used twice... Maybe we should have a test helper here so that we can track how many times a warning is detected even if it's skipped?

File: src/core.js [hunk 3]
```diff
@@ -82,6 +83,14 @@ migrateWarnProp( jQuery.expr, "filters", jQuery.expr.pseudos,
 migrateWarnProp( jQuery.expr, ":", jQuery.expr.pseudos,
 	"jQuery.expr[':'] is deprecated; use jQuery.expr.pseudos" );
 
+// Prior to jQuery 3.1.1 there were internal refs so we don't warn there
+if ( jQueryVersionSince( "3.1.1" ) ) {
+	migrateWarnFunc( jQuery, "trim", function( text ) {
+		return text == null ? "" : trim.call( text );
```
Comment (dmethvin) [inline src/core.js]: The [3.x code](https://github.com/jquery/jquery/blob/3.4.1/src/core.js#L271) uses `(text + "")` to ensure the argument is stringified in case it's something unusual. Seems like we'd want to do that here too in order for any edge cases to work.

File: test/core.js (hunk 2 above)
Comment (dmethvin) [reply test/core.js]: I'm fine with separate tests really. In real code once you know you're looking for `.trim` you can just search for that.

File: src/core.js (hunk 3 above)
Comment (dmethvin) [reply src/core.js]: Oh duh, we don't need it here!

Comment (mgol) [reply src/core.js]: Hmm, maybe I should use the 3.x version to keep compat with Android 4.0... After all, people can use jQuery 3.x with Migrate 3.x to prepare for a future jQuery 4.x upgrade even if they’re not ready to drop browsers not supported by jQuery 4.x yet.

Comment (dmethvin) [reply src/core.js]: Yeah, I agree with that. We have the code there and don't care about bytes in Migrate so we might as well use the same code.

Comment (mgol) [reply src/core.js]: PR updated.

**Comment by mgol** [comment]
@dmethvin I forgot about an entry in warnings.md, oops! Added, please have another look.

File: test/core.js (hunk 2 above)
Comment (mgol) [reply test/core.js:378]: @dmethvin I submitted #332 acting on the idea proposed in my comment above.
